### PR TITLE
refactor: move Gitlab and GitlabList to gitlab/client.py

### PR DESCRIPTION
**Pull request by JohnVillalovos**
Move the classes Gitlab and GitlabList from gitlab/__init__.py to the
newly created gitlab/client.py file.

Update one test case that was depending on requests being defined in
gitlab/__init__.py

### CODE REVIEW

**Review by nejch — COMMENTED** [review]
Thanks a lot @JohnVillalovos!

I think you had this mind already if I remember @max-wittig WDYT? It can be split more in a later PR if it balloons.. maybe the http requests stuff could go in its own module at some point.

File: gitlab/__init__.py [hunk 1]
```diff
@@ -32,6 +27,7 @@
     __title__,
     __version__,
 )
+from gitlab.client import Gitlab, GitlabList
 from gitlab.const import *  # noqa
 from gitlab.exceptions import *  # noqa
 from gitlab import utils  # noqa
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply gitlab/__init__.py:33]: I can remove them

File: gitlab/client.py [hunk 2]
```diff
@@ -0,0 +1,865 @@
...
+
+import gitlab.config
+from gitlab.__version__ import (
+    __author__,
+    __copyright__,
+    __email__,
+    __license__,
+    __title__,
+    __version__,
+)
```
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply gitlab/client.py:33]: Will do.

**Comment by JohnVillalovos** [comment]
I've updated it and hopefully addressed all the feedback. Thank you.

File: gitlab/__init__.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (JohnVillalovos) [reply gitlab/__init__.py:33]: As a note when I removed:

```
from gitlab import utils
```

The functional CLI test failed. So I put it back in.

[the remark this answers is not in the document]
Comment (nejch) [reply gitlab/__init__.py:33]: I suspect those might have just been some flaky functional tests, but I agree. we can keep it for now

**Review by nejch — APPROVED** [review]
Sorry @JohnVillalovos for the delay, could you just rebase? It's just this one added line https://github.com/python-gitlab/python-gitlab/commit/2bb16fac18a6a91847201c174f3bf1208338f6aa#diff-b0f8dbe3635cab16df9ce4bef4498422dbc0e6ad816820e00637f8f4e4d3bed0R146

Otherwise looks nice to have a clean init, unless @max-wittig has any more comments :)

**Comment by JohnVillalovos** [comment]
> 
> 
> Sorry @JohnVillalovos for the delay, could you just rebase? It's just this one added line [2bb16fa#diff-b0f8dbe3635cab16df9ce4bef4498422dbc0e6ad816820e00637f8f4e4d3bed0R146](https://github.com/python-gitlab/python-gitlab/commit/2bb16fac18a6a91847201c174f3bf1208338f6aa#diff-b0f8dbe3635cab16df9ce4bef4498422dbc0e6ad816820e00637f8f4e4d3bed0R146)
> 
> Otherwise looks nice to have a clean init, unless @max-wittig has any more comments :)

Done I believe. But please double check. I believe everything in the two classes (Gitlab and GitlabList) have been moved correctly to the new file from commit d4e7a031eb64ecba09f2547bd7803f2cceb7558b

**Comment by nejch** [comment]
> > Sorry @JohnVillalovos for the delay, could you just rebase? It's just this one added line [2bb16fa#diff-b0f8dbe3635cab16df9ce4bef4498422dbc0e6ad816820e00637f8f4e4d3bed0R146](https://github.com/python-gitlab/python-gitlab/commit/2bb16fac18a6a91847201c174f3bf1208338f6aa#diff-b0f8dbe3635cab16df9ce4bef4498422dbc0e6ad816820e00637f8f4e4d3bed0R146)
> > Otherwise looks nice to have a clean init, unless @max-wittig has any more comments :)
> 
> Done I believe. But please double check. I believe everything in the two classes (Gitlab and GitlabList) have been moved correctly to the new file from commit [d4e7a03](https://github.com/python-gitlab/python-gitlab/commit/d4e7a031eb64ecba09f2547bd7803f2cceb7558b)

Thanks, just checked with `git diff --color-moved=dimmed-zebra origin/master` and looks good.

**Pull request merged**